File: GUI/src-tauri/src/models/role.rs
#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct Role {
    pub id: String,
    pub name: String,
    pub icon: String,
    pub color: String,
    pub goal: String,
    pub personality_prompt: String,
    pub status: String,
    pub energy: i32,
    pub skills_config: String,
    pub proactivity_level: String,
    pub archived_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateRoleInput {
    pub name: String,
    pub icon: Option<String>,
    pub color: Option<String>,
    pub goal: Option<String>,
}

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateRoleInput {
    pub name: Option<String>,
    pub icon: Option<String>,
    pub color: Option<String>,
    pub goal: Option<String>,
    pub personality_prompt: Option<String>,
}

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateRoleSkillsInput {
    pub find_skills: bool,
    pub skill_creator: bool,
}

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ButlerSkillsConfig {
    pub find_skills: bool,
    pub skill_creator: bool,
}

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateRoleProactivityInput {
    pub proactivity_level: String,
}

File: egosync-app/src/types/role.ts
export type ProactivityLevel = 'passive' | 'moderate' | 'proactive';

export interface Role {
  id: string;
  name: string;
  icon: string;
  color: string;
  goal: string;
  personalityPrompt: string;
  status: 'active' | 'archived';
  energy: number;
  skillsConfig: string;
  proactivityLevel: ProactivityLevel;
  archivedAt: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRoleInput {
  name: string;
  icon?: string;
  color?: string;
  goal?: string;
}

export interface UpdateRoleInput {
  name?: string;
  icon?: string;
  color?: string;
  goal?: string;
  personalityPrompt?: string;
}

export interface UpdateRoleSkillsInput {
  findSkills: boolean;
  skillCreator: boolean;
  enabledSkillIds?: string[];
}

export interface RoleSkillsConfig {
  findSkills: boolean;
  skillCreator: boolean;
  enabledSkillIds: string[];
}

export interface ButlerSkillsConfig {
  findSkills: boolean;
  skillCreator: boolean;
  enabledSkillIds: string[];
}

export interface UpdateRoleProactivityInput {
  proactivityLevel: ProactivityLevel;
}
